src/sentry/snuba/metrics/datasource.py

```diff
@@ -542,46 +541,54 @@ def _get_group_limit_filters(
     if not metrics_query.groupby or not results:
         return None
 
-    # Translate the groupby fields of the query into their tag keys because these fields
-    # will be used to filter down and order the results of the 2nd query.
-    # For example, (project_id, transaction) is translated to (project_id, tags[3])
-    keys = tuple(
-        FIELD_ALIAS_MAPPINGS[metric_groupby_obj.name]
-        if metric_groupby_obj.name in FIELD_ALIAS_MAPPINGS
-        else resolve_tag_key(use_case_id, metrics_query.org_id, metric_groupby_obj.name)
-        if metric_groupby_obj.name not in FIELD_ALIAS_MAPPINGS.values()
-        else metric_groupby_obj.name
+    # Creates a mapping of groupBy fields to their equivalent SnQL
+    key_to_condition_dict: Dict[Groupable, Any] = OrderedDict()
+    for metric_groupby_obj in metrics_query.groupby:
+        key_to_condition_dict[
+            metric_groupby_obj.name
+        ] = SnubaQueryBuilder.generate_snql_for_groupby_field(
+            metric_groupby_obj=metric_groupby_obj,
+            use_case_id=use_case_id,
+            org_id=metrics_query.org_id,
+            projects=Project.objects.get_many_from_cache(metrics_query.project_ids),
+            is_column=True,
+        )
+
+    aliased_group_keys: Tuple[str] = tuple(
+        metric_groupby_obj.alias
         for metric_groupby_obj in metrics_query.groupby
+        if metric_groupby_obj.alias is not None
     )
-    aliased_group_keys = tuple(
-        metric_groupby_obj.alias for metric_groupby_obj in metrics_query.groupby
-    )
-
     # Get an ordered list of tuples containing the values of the group keys.
     # This needs to be deduplicated since in timeseries queries the same
     # grouping key will reappear for every time bucket.
     values = list(
         OrderedDict((tuple(row[col] for col in aliased_group_keys), None) for row in results).keys()
     )
     conditions = [
-        Condition(Function("tuple", [Column(k) for k in keys]), Op.IN, Function("tuple", values))
+        Condition(
+            Function("tuple", list(key_to_condition_dict.values())),
+            Op.IN,
+            Function("tuple", values),
+        )
     ]
-
     # In addition to filtering down on the tuple combination of the fields in
     # the group by columns, we need a separate condition for each of the columns
     # in the group by with their respective values so Clickhouse can filter the
     # results down before checking for the group by column combinations.
     values_by_column = {
         key: list({row[aliased_key] for row in results})
-        for key, aliased_key in zip(keys, aliased_group_keys)
+        for key, aliased_key in zip(list(key_to_condition_dict.keys()), aliased_group_keys)
```

Comment:
```suggestion
        for key, aliased_key in zip(key_to_condition_dict.keys(), aliased_group_keys)
```
i just checked: `zip(dict.keys(), aliased_group_keys)` works, so list is redundant.

Reply:
agreed